shared/js/background/classes/ad-click-attribution-policy.js

```diff
@@ -174,3 +207,54 @@ export class AdClick {
         return this.hasNotExpired()
     }
 }
+
+/**
+ * @param {number} tabId
+ * @param {object} allowlist
+ **/
+export class AdClickDNR {
+    constructor (tabId, allowlist) {
+        this.allowlist = allowlist
+        this.tabId = tabId
+        this.initiatorDomain = null
+        this.rule = generateDNRRule({
+            id: null,
+            priority: AD_ATTRIBUTION_POLICY_PRIORITY,
+            actionType: 'allow',
+            requestDomains: allowlist.map((entry) => entry.host)
+        })
+        this.rule.condition.tabIds = [tabId]
+        this.createInitialAdClickDNR()
+    }
+
+    /**
+     * Create initial tab scoped DNR not limited to an initiatorDomain
+     **/
+    async createInitialAdClickDNR () {
+        const ruleId = await getNextSessionRuleId()
+        this.rule.id = ruleId
+        chrome.declarativeNetRequest.updateSessionRules({ addRules: [this.rule] })
+    }
+
+    /*
+     * Update the tab DNR with initiator domain
+     */
+    addAdClickDNRInitiatorDomain (domain) {
+        if (!this.initiatorDomain) {
```

Comment:
Can you explain why not set this up front or when it's set in adClick?

Are we perhaps creating a lot of additional work by separating out the DNR class?

Reply:
I might be able to set it in adClick. Let me test that out.